int32[4] wheel_value